shared/js/views.jsx

```diff
@@ -977,67 +935,39 @@ loop.shared.views = (function(_, mozL10n) {
       );
     },
 
-    renderMediaWait: function() {
-      var msg = mozL10n.get("call_progress_getting_media_description",
-                                { clientShortname: mozL10n.get("clientShortname2") });
-      var utils = loop.shared.utils;
-      var isChrome = utils.isChrome(navigator.userAgent);
-      var isFirefox = utils.isFirefox(navigator.userAgent);
-      var isOpera = utils.isOpera(navigator.userAgent);
-      var promptMediaMessageClasses = classNames({
-        "prompt-media-message": true,
-        "chrome": isChrome,
-        "firefox": isFirefox,
-        "opera": isOpera,
-        "other": !isChrome && !isFirefox && !isOpera
-      });
-      return (
-        <div className="prompt-media-message-wrapper">
-        <p className={promptMediaMessageClasses}>
-          {msg}
-        </p>
-        </div>
-      );
-    },
-
     render: function() {
+      // TODO set class in the dom
       var remoteStreamClasses = classNames({
-        "remote": true,
-        // XXX akita-sidebar
-        // "focus-stream": !this.props.displayScreenShare
-        "focus-stream": false
+        "remote": true
       });
 
       var mediaWrapperClasses = classNames({
         "media-wrapper": true,
-        // XXX akita-sidebar
-        // "receiving-screen-share": this.props.displayScreenShare,
-        "receiving-screen-share": false,
         "showing-local-streams": this.props.localSrcMediaElement ||
           this.props.localPosterUrl,
         "showing-media-wait": this.props.showMediaWait,
-        "showing-remote-streams": this.props.remoteSrcMediaElement ||
-          this.props.remotePosterUrl || this.props.isRemoteLoading
+        "showing-remote-streams": this._shouldShowRemoteStream()
       });
 
-      // XXX akita-sidebar
       return (
         <div className="media-layout">
           <div className={mediaWrapperClasses}>
             <span className="self-view-hidden-message">
               {mozL10n.get("self_view_hidden_message")}
             </span>
-            <div className={remoteStreamClasses}>
-              <MediaView
-                displayAvatar={!this.props.renderRemoteVideo}
-                isLoading={this.props.isRemoteLoading}
-                mediaType="remote"
-                posterUrl={this.props.remotePosterUrl}
-                srcMediaElement={this.props.remoteSrcMediaElement} />
-              {this.state.localMediaAboslutelyPositioned ?
-                this.renderLocalVideo() : null}
-              {this.props.children}
-            </div>
+            {
+              this._shouldShowRemoteStream() ?
+              <div className={remoteStreamClasses}>
```

Comment:
This feels like its getting a bit confusing and difficult to maintain. I can see there being some hidden gotchas when it comes to css as well.

The main reason previously for moving the elements around was due to the different screen layouts and requirements (e.g. not always showing the screen share). As we've now dropped some of that complication, I'm thinking we should be able to simplify this. Maybe something like:

```js
<MediaView mediaType="remote" ... />
<MediaView mediaType="local" ... />
<MediaButtonsView .../>
<MediaWaitView ... />
```

I'm not sure, but I'm also wondering about whether or not we should move the decision of whether to render the contents of a component into the component itself, I think it would be nice, but I'm not sure if that would make sense or not. Maybe try it out, or see if @dmose has some thoughts?